cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -240,6 +227,20 @@ func start(log log.Component, config config.Component, forwarder defaultforwarde
 		pkglog.Info("Orchestrator explorer is disabled")
 	}
 
+	// Initialize remote configuration - needs to be done after the cluster ID has been created
+	// (if it's going to be) by the cluster-agent so that info can be reported back to the backend.
+	var rcClient *remote.Client
+	if pkgconfig.Datadog.GetBool("remote_configuration.enabled") {
+		var err error
+		rcClient, err = initializeRemoteConfig(mainCtx)
```

Comment:
We can. Originally we wanted to get this in for the 7.45 RC releases as a bugfix so I was trying to address this with as few changes as possible. Now that we're not doing that though we can make broader changes.

The awkward issue is this client is shared by all agent sub-processes that have remote config products, and only the cluster-agent is going to have a cluster ID parameter to report back. I can probably limit that with something like `NewRemoteConfigClientWithClusterID()` so that other agent sub-processes don't have to change and expand the scope of the PR and the cluster-agent just uses this special constructor.